### PR TITLE
Rename SuccessURLAllowedHostsMixin to RedirectURLMixin and hint missing attributes

### PR DESCRIPTION
**Pull request by si14**
Just a simple stub for `contrib.auth.views.RedirectURLMixin`.

I noticed that it was in the todolist and I needed it, so here it is.

### CODE REVIEW

File: django-stubs/contrib/auth/views.pyi [hunk 1]
```diff
@@ -9,6 +9,15 @@ from django.views.generic.edit import FormView
...
+class RedirectURLMixin:
+    next_page: str | None
+    redirect_field_name: str
+    success_url_allowed_hosts: set[str]
+    def get_success_url(self) -> str: ...
+    def get_redirect_url(self) -> str: ...
+    def get_success_url_allowed_hosts(self) -> set[str]: ...
+    def get_default_redirect_url(self) -> str: ...
+
 class SuccessURLAllowedHostsMixin:
```
Comment (intgr) [inline django-stubs/contrib/auth/views.pyi]: Some more changes look warranted...

`SuccessURLAllowedHostsMixin` no longer exists in Django 4.2, `LoginView` and `LogoutView` should inherit from `RedirectURLMixin` instead.

**Review by intgr — CHANGES_REQUESTED** [review]
Please remove `SuccessURLAllowedHostsMixin` and replace it with the new `RedirectURLMixin`.

**Comment by si14** [comment]
@intgr fixed, all tests are green

**Review by intgr — APPROVED** [review]
Thank you!

**Pull request merged**